pkg/wallet/internal/actions/wallet_internalize_action.go

```diff
@@ -0,0 +1,88 @@
+package actions
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bsv-blockchain/go-sdk/transaction"
+	"github.com/bsv-blockchain/go-sdk/transaction/template/p2pkh"
+	sdk "github.com/bsv-blockchain/go-sdk/wallet"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/brc29"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
+)
+
+// ValidateWalletInternalizeAction performs wallet-specific validation for internalize actions
+func ValidateWalletInternalizeAction(ctx context.Context, keyDeriver *sdk.KeyDeriver, args *wdk.InternalizeActionArgs) error {
+	beef, txIDHash, err := transaction.NewBeefFromAtomicBytes(args.Tx)
+	if err != nil {
+		return fmt.Errorf("failed to create atomic beef from bytes: %w", err)
+	}
+
+	tx := beef.FindAtomicTransactionByHash(txIDHash)
+	if tx == nil {
+		return fmt.Errorf("atomic beef error: transaction with hash %s not found", txIDHash)
+	}
+
+	for _, output := range args.Outputs {
+		if err := validateOutput(keyDeriver, *output, tx); err != nil {
+			return fmt.Errorf("output validation failed: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// validateOutput validates a single output based on its protocol
+func validateOutput(keyDeriver *sdk.KeyDeriver, output wdk.InternalizeOutput, tx *transaction.Transaction) error {
+
+	txOutput := tx.Outputs[output.OutputIndex]
+
+	switch output.Protocol {
+	case wdk.WalletPaymentProtocol:
+		return validateWalletPaymentOutput(keyDeriver, output, txOutput)
+	case wdk.BasketInsertionProtocol:
+		return validateBasketInsertionOutput()
+	default:
+		return fmt.Errorf("unexpected protocol: %s", output.Protocol)
+	}
+}
+
+// validateWalletPaymentOutput validates a wallet payment output using BRC-29
+func validateWalletPaymentOutput(keyDeriver *sdk.KeyDeriver, output wdk.InternalizeOutput, txOutput *transaction.TransactionOutput) error {
+
+	payment := output.PaymentRemittance
+
+	keyID := brc29.KeyID{
+		DerivationPrefix: string(payment.DerivationPrefix),
+		DerivationSuffix: string(payment.DerivationSuffix),
+	}
+
+	if keyDeriver == nil {
+		return fmt.Errorf("wallet keyDeriver is not available")
+	}
+
+	senderPubKey := brc29.PubHex(payment.SenderIdentityKey)
+
+	expectedAddress, err := brc29.AddressForSelf(senderPubKey, keyID, keyDeriver)
+	if err != nil {
+		return fmt.Errorf("failed to generate expected BRC-29 address: %w", err)
+	}
+
+	expectedLockScript, err := p2pkh.Lock(expectedAddress)
+	if err != nil {
+		return fmt.Errorf("failed to create expected P2PKH locking script: %w", err)
+	}
```

Comment:
You should use brc29.Lock, instead of creating address and p2pkh.Lock, like:

```suggestion
	expectedLockScript, err := brc29.Lock(brc29.PubHex(payment.SenderIdentityKey), keyID, keyDeriver)
	if err != nil {
		return fmt.Errorf("failed to create expected locking script: %w", err)
	}
```

Reply:
Actually, the Lock only performs the method as per counter party private key perspective currently.
I can consolidate the key deriver validation and the senderPubKey variable though.

We could consider creating - LockForSelf method, but might be too much for a single use case. WDYT?

Reply:
actually, I see that we're using it in 2 places, so maybe it would be useful to have function LockForSelf.

Finally, we have real use case for adding it.